### PR TITLE
Replace `envoy.router` with `envoy.filters.http.router`

### LINKED ISSUE #29737: Replace `envoy.router` with `envoy.filters.http.router`
`envoy.router` is widely used by envoyfilters, we should update all of them.

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fix: #29737

### CODE REVIEW

File: tests/integration/pilot/testdata/upgrade/1.8.0-install.yaml
Comment (howardjohn) [inline tests/integration/pilot/testdata/upgrade/1.8.0-install.yaml:265]: this is intentionally a snapshot of the old install charts to test upgrade so it shouldn't change

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go:1426]: This is left intentionally with the deprecated name for the test to verify that listener patch works with both deprecated and new canonical names

**Comment by ramaraochavali** [comment]
Just put a hold to avoid accidental merge - We need to revert the intentional usages

File: tests/integration/pilot/testdata/upgrade/1.8.0-install.yaml
Comment (hzxuzhonghu) [reply tests/integration/pilot/testdata/upgrade/1.8.0-install.yaml:265]: Will revert

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go:1426]: Will revert

**Review by mandarjog — COMMENTED** [review]
are we maintaining equivalence between envoy.filters.http.router and envoy.router for the EnvoyFilter api ? None of that has changed here it looks like.

**Comment by ramaraochavali** [comment]
>  are we maintaining equivalence between envoy.filters.http.router and envoy.router for the EnvoyFilter api ? None of that has changed here it looks like.

Yes. We are maintaining the equivalence and this PR has not changed that.

**Review by ramaraochavali — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
> >are we maintaining equivalence between envoy.filters.http.router and envoy.router for the EnvoyFilter api ? None of that has changed here it looks like.

>Yes. We are maintaining the equivalence and this PR has not changed that.

We can document in 1.9 that the legacy names are not supported anymore, we never say envoyfilter is backward compatible

**Comment by mandarjog** [comment]
@hzxuzhonghu we can document it, however we have to have code that deals with it. I think it is ok to explicitly warn  that old names are not supported in this release and then fail validation is a subsequent release. We cannot silently ignore the option, which is what will happen if we simply stop recognizing it.

**Comment by howardjohn** [comment]
I believe we already warn - which is why the telemetry v2 part of this
change is great since our default install shouldn't have warnings

**Comment by mandarjog** [comment]
Yeah, good to not have warnings by default. In 1.9 will not use this method to install telemetry, though we will have to work with older artifacts.

**Comment by ramaraochavali** [comment]
We already added a warning in the upgrade doc https://istio.io/latest/news/releases/1.8.x/announcing-1.8/upgrade-notes/#use-the-new-filter-names-for-envoyfilter for 1.8

**Comment by hzxuzhonghu** [comment]
Thanks for the info. Removing support is a separate thing, we can do that in 1.9.

**Comment by hzxuzhonghu** [comment]
@istio/wg-docs-maintainers Need your approval

**Review by ericvn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@howardjohn Can you approve

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@mandarjog need owners from telemetry to approve

**Comment by hzxuzhonghu** [comment]
ping @istio/wg-policies-and-telemetry-maintainers

**Review by mandarjog — APPROVED** [review]
Lgtm

**Pull request merged**
